Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/ChunkedHttpDownloader.cs

```diff
@@ -210,14 +227,50 @@ private bool TryDownload(ResourceUrl url, ChunkedFileStream fileStream, Cancella
             }
         }
 
+        public BytesRange CalculateContainingChunksRange(BytesRange range)
+        {
+            long chunkSize = _chunksData.ChunkSize;
+            long bottom = (range.Start / chunkSize) * chunkSize;
+
+            if (range.End == -1)
+            {
+                return new BytesRange(bottom, -1);
+            }
+
+            long top = (range.End / chunkSize) * chunkSize;
```

Comment:
Wouldn't it be better to?:
`long top = Math.Min((range.End / chunkSize) * (chunkSize + 1), range.End);`